### ISSUE TITLE
YoloV4 trains "too quickly"

### ISSUE THREAD

**Issue by WefPok** [AlexeyAB/darknet#7609]
I'm training default yolov4-tiny. But the problem is that it has loss of 0.2 after 200 iterations and it provides no detections.
Tried to train on different datasets. F.E. Open images - one class (Apple)(with labels correctly converted, which I checked through Yolo Mark).
Info: 
![image](https://user-images.githubusercontent.com/46328702/114995216-42f45280-9ea6-11eb-8e03-b8f65b51baae.png)
The only thing that I changed in .cfg is resolution. (ofc changed classes and filters in YOLO layers ). Tried to train with -show_imgs. Bounding boxes are correct.

**Comment by stephanecharette** [comment]
> But the problem is that it has loss of 0.2 after 200 iterations and it provides no detections.

The *MINIMUM* iterations for 1 class is 6000.  And it goes up from there depending on the number of images and the number of classes.

**Comment by WefPok** [comment]
> > But the problem is that it has loss of 0.2 after 200 iterations and it provides no detections.
> 
> The _MINIMUM_ iterations for 1 class is 6000. And it goes up from there depending on the number of images and the number of classes. This is explained in the readme file in the project's root directory.

Yeah, I understand that, but usually loss is gradually decreasing, in this case, i got loss 0.0000 after 600 iterations already. Is it normal behaviour? Because I've never faced such problem. Could you please specify where is it described?

**Comment by stephanecharette** [comment]
> Could you please specify where is it described?

In the readme:  https://github.com/AlexeyAB/darknet

Search for `6000` or `max_batches`.

> Is it normal behaviour?

I have no idea.  I haven't seen your dataset, your annotations, your config, etc.  But if you have a single class to detect "apple" I would say you've probably done something wrong if you're now at 3000 iterations and it cannot detect an apple.

What command are you running?  Both to train, and then to test if it can detect anything?  What size are your images?  What size are the apples in the images?  And what size is your network?

**Comment by WefPok** [comment]
> What command are you running? Both to train, and then to test if it can detect anything? What size are your images? What size are the apples in the images? And what size is your network?

Command: 
```!./darknet detector train data/obj.data cfg/yolov4-obj.cfg yolov4-tiny.conv.29 -dont_show```
Apples of all sizes are present, image sizes also different. The network is 416x416. 
I took yolov4-tiny-custom.cfg, Changed: 

-  max_batches = 6000
- steps = 4800, 5400
- classes = 1 (Yolo Layer)
- filters = 18 (in the layers above yolo layers)
And that's it. I've trained many models before, a week ago everything was OK, but now smth goes wrong, I'm not able to train, because of this problem. Like, why would the loss decrease SO fast?
